### ISSUE TITLE
Signign with Mac Developer certificates

### ISSUE THREAD

**Issue by vtardia** [electron-userland/electron-builder#812]
- **Version**: 7.11.3

- **Target**: macOS

Hi, with electron-builder 3 I was able to sign my app previews with my "Mac Developer" certificate. I've updated to 7 and I'm no longer able to do this, and I have to disable signing and manually sign with `codesign`.

Why? Can you put back this feature?

**Comment by develar** [comment]
> Why? Can you put back this feature?

You must not use this certificate type. Please see #740 and https://developer.apple.com/library/content/documentation/IDEs/Conceptual/AppDistributionGuide/MaintainingCertificates/MaintainingCertificates.html

**Mac Developer – Used to enable certain app services for a Mac app during development and testing.**

So, you must use `Developer ID Application` — **Used to sign a Mac app before distributing it outside the Mac App Store.**.

**Comment by vtardia** [comment]
I'm familiar with both #740 and Apple guidelines, but I still need to be able to use Mac Developer certificates for development and testing, and let my client(s) use Developer IDs for distribution...

**Comment by develar** [comment]
Could you please explain me, why you cannot use `Developer ID` cert? Why `Mac Developer ` is required?

**Comment by vtardia** [comment]
Because as far as I know "Developer ID" is currently available only for paid plans

**Comment by develar** [comment]
> as far as I know "Developer ID" is currently available only for paid plans

Sorry, I don't understand. Any Apple cert requires Apple Developer account — in any case you have to pay for developer account and any cert is after that is free for you.

**Comment by vtardia** [comment]
With an Apple ID you can have free membership and a Mac Developer certificate for testing https://developer.apple.com/support/compare-memberships/. I develop and test with my free account and certificate, and my clients sign with their paid membership certificates.

**Comment by develar** [comment]
Certificate will be allowed.

**Comment by vtardia** [comment]
Thank you very much @develar!

**Comment by cumajkeee** [comment]
Hi @develar, does it implemented/fixed? Still can't sign with "Mac Developer" cert.

Cert is correctly listed here: `security find-identity -v`

**Comment by develar** [comment]
@cumajkeee What version of electron-builder do you use? 
Please set env DEBUG=electron-builder and attach log of the terminal output.

**Comment by cumajkeee** [comment]
@develar: 
v=11.2.3

Tue, 10 Jan 2017 14:12:26 GMT electron-builder Executing security find-identity -v
Tue, 10 Jan 2017 14:12:26 GMT electron-builder Executing security find-identity -v -p codesigning
Error: Identity name "Mac Developer: email@email.com (URxxxxxxxx)" is specified, but no valid identity with this name in the keychain
    at /{path}/electron/node_modules/electron-builder/src/codeSign.ts:226:13
From previous event:

![screenshot 2017-01-10 17 16 11](https://cloud.githubusercontent.com/assets/1834053/21809550/4acf12c4-d759-11e6-89e4-03b5b906e011.jpg)

**Comment by develar** [comment]
@cumajkeee if you specify identity manually, you should specify only name, without type prefix.

**Comment by ericzhangjx** [comment]
@develar 
Actually doing so could also cause Error:
```
Packaging for darwin x64 using electron 1.4.14 to dist/Mac
Error: Identity name "email@email.com (QWERT12345)" is specified, but no valid identity with this name in the keychain
    at /Users/Eric/WorkSpace/PROJECT/node_modules/electron-builder/src/codeSign.ts:226:13
```
node: v=7.4.0
npm: v=4.0.5
electron-builder: v=11.2.4

**Comment by ericzhangjx** [comment]
#1103

**Comment by ericzhangjx** [comment]
#1103 
Seems to have solved the problem (by removing an unnecessary exception inside `findIdentity()`).

**Comment by develar** [comment]
Fixed in upcoming 11.2.4 (thanks to @ericzhangjx).

**Comment by cumajkeee** [comment]
Thanks @ericzhangjx  and @develar!
